pkg/tracing/tracing.go

```diff
@@ -52,16 +54,36 @@ func Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption)
 	return TracerFromCtx(ctx).Start(ctx, spanName, opts...)
 }
 
-const SpanAttrWarpforgeErrorCode = "warpforge.error.code"
-const SpanAttrWarpforgeFormulaId = "warpforge.formula.id"
-const SpanAttrWarpforgePackId = "warpforge.pack.id"
-
 // SetSpanError is a helper function to set the span error based on a wfapi.Error
 func SetSpanError(ctx context.Context, err wfapi.Error) {
 	e := err.(*wfapi.ErrorVal)
 	span := trace.SpanFromContext(ctx)
 	span.SetAttributes(
-		attribute.String(SpanAttrWarpforgeErrorCode, e.Code()),
+		attribute.String(AttrKeyWarpforgeErrorCode, e.Code()),
 	)
 	span.SetStatus(codes.Error, e.Error())
 }
+
+func EndWithStatus(span trace.Span, err error) {
+	if err == nil {
+		span.SetStatus(codes.Ok, "")
+		return
+	}
+	span.SetStatus(codes.Error, err.Error())
+	span.End()
+}
+
+// PrintableAttribute creates an attribute.KeyValue with only printable characters from the value
+func PrintableAttribute(key string, value string) attribute.KeyValue {
+	return attribute.String(key, Printable(value))
+}
+
+// Printable returns the input with only printable characters as defined by unicode.IsGraphic
+func Printable(s string) string {
+	return strings.Map(func(r rune) rune {
+		if unicode.IsGraphic(r) {
+			return r
+		}
+		return -1
+	}, s)
+}
```

Comment:
I wonder if perhaps you might be looking for https://pkg.go.dev/strconv#QuoteToGraphic here?

Although also I should note that the only place I saw this being used was on step names, and those are not meant to permit totally wild nongraphical characters either.  (Enforcement, admittedly, probably not-yet-implemented.)  So this might be slightly overkill / adding enforcement against wild stepnames might be a better investment.

Reply:
Hmm, I don't really want the surrounding quotes though. I do generally think we are going to want some kind of user-input validation/protection but I'm not sure what that looks like exactly. I was thinking that I might run a similar filter on the logging output in the hopes of fixing the tty bug.